cli/ingress-controller/flags.go

```diff
@@ -289,49 +261,29 @@ func parseFlags() (cliConfig, error) {
 
 	// Kong connection details
 	kongAdminURL := defaultKongAdminURL
-	oldURL := viper.GetString("kong-url")
-	newURL := viper.GetString("kong-admin-url")
-	if oldURL != "" {
-		kongAdminURL = oldURL
-	}
-	if newURL != defaultKongAdminURL {
-		kongAdminURL = newURL
+	flagURL := viper.GetString("kong-admin-url")
+	if flagURL != defaultKongAdminURL {
+		kongAdminURL = flagURL
 	}
```

Comment:
This should be simplified further as the complicated logic is going away:

```go
config.KongAdminURL = viper.GetString("kong-admin-url")
```

Reply:
Not blocking:

Food for thought: we might want to drop all the binding (`config.Xxx = viper.GetXxx(Xxx)`) logic by unmarshaling from viper directly into a `config` struct:

https://godoc.org/github.com/spf13/viper#UnmarshalExact

It would be even easier if we weren't using viper (because we could declare `StringVar`, `IntVar` instead of `String`, `Int` which would write directly to the `config` struct) but we need viper as long as we want to support the flag/envvar dualism of config values.

Reply:
tracked in https://github.com/Kong/kubernetes-ingress-controller/issues/868